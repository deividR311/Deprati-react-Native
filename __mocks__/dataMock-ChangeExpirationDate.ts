export const DataMock_ChangeExpirationDate = {
  TEXT_NEWDATE: 'Selecciona el día',
  infoCreditBalance: {
    clientName: 'PATRICIA L. DE UGALDE',
    cardNumber: '1xxxxx6',
    affiliateDate: '2000',
    totalAmount: '5.500,00',
    amountSpent: '724,35',
    aditional: '00',
    availableAmount: '4.775,65',
    pastDueDays: 0,
    nextPayDay: 'Abril 14 de 2023',
    nextAmountToPay: '105,08',
    numeroTarjetaAdicionalDisplay: '14XXX86 00',
    clientFirstName: 'Prueba'
  },
  dataCurrentDate: {
    cuttingDay: 14,
    dueDate: '20230514',
    day: '14'
  },
  dataChangeDate: null,
  selectedDay: '2',
  isAcceptTerms: false,
  showWarning: false,
  isLoadingChangeDate: false,
  isLoadingCurrentDate: false,
  showSucces: false,
  showErrorSucces: false,
  expirationDateContent: {
    legend:
      'No puedes cambiar tu fecha máxima de pago si tienes una deuda pendiente y <b>este cambio se hará efectivo a partir del próximo mes.</b>',
    popup_title:
      '¿Estás seguro de cambiar la fecha máxima de pago de tu crédito De Prati?',
    popup_paragraphs: [
      'Ten en cuenta:',
      '1. Esta acción no se puede deshacer y el cambio se hará efectivo a partir de tu próximo mes.',
      '2. Si haces el cambio de fecha <b>deberás pagar la cuota del mes en curso</b> y este cambio se hará efectivo a partir del próximo mes.'
    ],
    popup_accept_action: 'SI, ESTOY DE ACUERDO',
    popup_cancel_action: 'NO, CANCELAR'
  },
  selectRef: {
    current: {
      reset: jest.fn(),
      openDropdown: jest.fn(),
      closeDropdown: jest.fn()
    }
  },
  setSelectedDay: jest.fn(),
  setIsAcceptTerms: jest.fn(),
  handleChangeDate: jest.fn(),
  setShowWarning: jest.fn(),
  handleCloseSuccess: jest.fn(),
  handleCloseError: jest.fn(),
  handleCloseWarning: jest.fn()
};
