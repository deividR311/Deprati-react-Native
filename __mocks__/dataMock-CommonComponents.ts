export const DataMock_CustomProps = {
  url: 'url',
  style: {},
  content: 'qwerty',
  uid: 'uid',
  typeCode: 'qwerty',
  modifiedTime: 'qwerty',
  name: 'qwerty',
  container: true,
  rotatingImagesComponentsList: 'DepratiBrandsRotatingImages',
  childrenComponentsList: ['DepratiBrandsRotatingImages'],
  childrenComponents: {
    DepratiBrandsRotatingImages: {
      uid: 'DepratiBrandsRotatingImages',
      typeCode: 'RotatingImagesComponent',
      modifiedTime: '2023-04-28T12:59:58+0000',
      name: 'DePrati Dior Mobile Carousel',
      container: 'false',
      resolutionEnum: 'MOBILE',
      effect: 'FADE',
      banners: 'DiorBannerComponent2 BannerCarouselDiorVideo_1',
      timeout: '6',
      childrenComponentsList: [
        'DiorBannerComponent2',
        'BannerCarouselDiorVideo_1'
      ]
    }
  }
};

export const DataMock_BillingAddressComponent = {
  style: {},
  onSelected: () => {},
  emptyStyle: {},
  showTitle: true,
  enableHandleEnableButton: true,
  onIsEnable: () => {},
  paymentAddress: {
    defaultAddress: false,
    email: 'string',
    firstName: 'string',
    formattedAddress: 'string',
    id: 'string',
    line1: 'string',
    line2: 'string',
    phone: 'string',
    phonePreffix: 'string',
    shippingAddress: false,
    visibleInAddressBook: false
  },
  paymentCredit: {
    showAlerAddressPayment: false,
    setShowAlerAddressPayment: () => {}
  },
  styleSecundaryButton: false
};

export const DataMock_useBillingAddress = {
  addressSelected: {
    defaultAddress: false,
    firstName: 'firstName',
    id: 'qwertystring',
    line1: 'qwerty',
    line2: 'qwerty',
    phone: 'qwerty',
    phonePreffix: 'qwerty'
  },
  isLoadingByDeleting: false,
  addressPaymentList: {
    default: null,
    others: [],
    existsAddress: false,
    loading: false
  },
  isLoading: false,
  buttonChangeData: false,
  handleChangeData: jest.fn(),
  changeAddress: {
    handleSelected: jest.fn(),
    addressSelected: undefined
  },
  handleDelete: jest.fn(),
  handleAddEdit: jest.fn(),
  handleBillingAddressList: () => {
    return [
      {
        defaultAddress: false,
        email: 'string',
        firstName: 'string',
        formattedAddress: 'string',
        id: 'string',
        line1: 'string',
        line2: 'string',
        phone: 'string',
        phonePreffix: 'string',
        shippingAddress: false,
        visibleInAddressBook: false
      }
    ];
  }
};

export const DataMock_IncreaseCreditLimit = {
  rectangularSize: 232,
  value: 1000,
  maxValue: 2000,
  onChangeCreditLimit: jest.fn()
};

export const DataMock_Charts = {
  constentChart: {
    availableAmount: 100,
    amountSpent: 20,
    totalAmount: 200
  },
  ownerFullname: 'Custom customer customization',
  invertValues: true,
  showDisplayCard: true
};
