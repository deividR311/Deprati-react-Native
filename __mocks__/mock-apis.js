export var mockhookApi = [
  jest.fn().mockResolvedValue(),
  {
    data: {
      data: {}
    },
    isLoading: false,
    isSuccess: true,
    isError: false,
    reset: jest.fn()
  }
];

jest.mock('../src/infrastucture/apis/user/user.api', () => ({
  useForgottenpasswordRequest: () => [
    jest.fn().mockResolvedValue({ data: true })
  ],
  useGetLazyInfoUserRequest: () => [
    jest.fn().mockResolvedValue({ data: true })
  ],
  useGetInfoUserRequest: () => [jest.fn().mockResolvedValue({ data: true })],
  useUpdateProfileRequest: () => [jest.fn().mockResolvedValue({ data: true })],
  useNewUserRequest: () => [jest.fn().mockResolvedValue({ data: true })]
}));

jest.mock('../src/infrastucture/apis/contentPage/contentPage.hook.ts', () => {
  const mockFn = jest.fn().mockReturnValue({
    loading: false,
    pageContent: {},
    getDataContent: jest.fn(),
    error: false
  });
  return {
    __esModule: true,
    default: mockFn,
    usePageContent: mockFn
  };
});

jest.mock('../src/infrastucture/apis/wishlist/wishlist.api', () => ({
  useLazyCustomerWishlistRequest: () => mockhookApi,
  useAddWishlistMutationRequest: () => mockhookApi,
  useRemoveWishlistMutationRequest: () => mockhookApi
}));

jest.mock('../src/infrastucture/apis/product/product.api.tsx', () => ({
  useReviewMutationRequest: () => mockhookApi,
  useProductMutationRequest: () => mockhookApi,
  useSearchMutationRequest: () => mockhookApi,
  useGetProductsRequest: () => mockhookApi
}));

jest.mock(
  '../src/infrastucture/apis/customer-orders/customer-orders.api',
  () => ({
    useLazyCustomerOrdersRequest: () => mockhookApi,
    useOrderDetailsRequest: () => mockhookApi,
    useOrderProductReviewRequest: () => mockhookApi
  })
);

jest.mock('../src/infrastucture/apis/contactless-payment', () => ({
  useContactlessTicketByIdRequest: () => mockhookApi,
  useContactlessTicketRequest: () => mockhookApi,
  useContactlessTicketsRequest: () => mockhookApi,
  useContactlessPurchaseConfirmRequest: () => mockhookApi,
  useContactlessPurchaseCancelRequest: () => mockhookApi,
  useRegisteTokenTicketRequest: () => mockhookApi
}));

jest.mock('../src/infrastucture/apis/customer-orders', () => ({
  useLazyCustomerOrdersRequest: () => mockhookApi,
  useOrderDetailsRequest: () => mockhookApi,
  useOrderProductReviewRequest: () => mockhookApi
}));

jest.mock('../src/infrastucture/apis/delivery-address', () => ({
  useLazyGetDeliveryTimesQuery: () => mockhookApi,
  useSetDeliveryAddressMutation: () => mockhookApi,
  useSetDeliveryModeMutation: () => mockhookApi,
  useAddressRemoveCartRequest: () => mockhookApi,
  useDeliveryOptionsRequest: () => mockhookApi
}));

jest.mock('../src/infrastucture/apis/shopping-cart/shopping-cart.api', () => ({
  useLazyGetDeliveryTimesQuery: () => mockhookApi,
  useLazyGetShoppingCartRequest: () => mockhookApi,
  useGetShoppingCartRequest: () => mockhookApi,
  useCreateShoppingCartRequest: () => mockhookApi,
  useAddToShoppingCartRequest: () => mockhookApi,
  useUpdateShoppingCartRequest: () => mockhookApi,
  useDeleteShoppingCartRequest: () => mockhookApi,
  useAddCouponRequest: () => mockhookApi,
  useRemoveCouponRequest: () => mockhookApi,
  useGifPackageShoppingCartRequest: () => mockhookApi,
  usePickUpShoppingCartRequest: () => mockhookApi
}));

jest.mock(
  '../src/infrastucture/apis/delivery-thirdparty/delivery-thirdparty.api',
  () => ({
    useGetProvincesMutation: () => mockhookApi,
    useGetCitiesMutation: () => mockhookApi,
    useGetThirdPartyAgenciesListMutation: () => mockhookApi,
    useSelectThirdPartyAgencyMutation: () => mockhookApi,
    useUpdateThirdPartyAgencyMutation: () => mockhookApi
  })
);

jest.mock('../src/infrastucture/apis/creditStatus/creditStatus.tsx', () => ({
  useDcStateCreditMutationrquest: () => mockhookApi
}));
jest.mock('../src/infrastucture/apis/checkMail/checkMail.tsx', () => ({
  useCheckMailMutationRequest: () => mockhookApi
}));
// jest.mock(
//   '../src/infrastucture/apis/contentService/contentService.api.tsx',
//   () => ({
//     useContentServiceMutation: () => mockhookApi,
//   }),
// )
jest.mock(
  '../src/infrastucture/apis/creditMovement/credit-movement.api.tsx',
  () => ({
    useAccountMovementsMutationRequest: () => mockhookApi,
    useAccountMovementsPeriodsRequest: () => [
      jest.fn().mockResolvedValue(),
      {
        movementPeriods: {
          data: {}
        },
        isLoading: false,
        isSuccess: true,
        isError: false
      }
    ]
  })
);
jest.mock('../src/infrastucture/apis/myReturns/returns.api.tsx', () => ({
  useCheckMailMutationRequest: () => mockhookApi,
  useLazyReturnsListRequest: () => mockhookApi,
  useLazyReturnableOrdersRequest: () => mockhookApi,
  useLazySearchReturnsOrdersRequest: () => mockhookApi,
  useLazyDetailReturnRequest: () => mockhookApi,
  useEnterReturnRequest: () => mockhookApi
}));

jest.mock('../src/presentation/screens/Splash/hooks/useSplash.hook', () => ({
  useSplash: jest.fn().mockReturnValue({
    hasError: false
  })
}));
