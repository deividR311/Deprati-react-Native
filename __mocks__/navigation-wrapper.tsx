import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { COLORS } from '../src/application/common';
import { ProvideWrapper } from './provider-wrapper';

jest.useFakeTimers();
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

const Stack = createNativeStackNavigator();

export const NavigationWrapper = ({ children, screen, initialParams }: any) => (
  <ProvideWrapper>
    <NavigationContainer>
      <Stack.Navigator
        initialRouteName="default"
        screenOptions={{
          headerShown: true,
          headerShadowVisible: false,
          headerTitleAlign: 'left',
          contentStyle: { backgroundColor: COLORS.WHITE }
        }}>
        {screen && (
          <Stack.Screen
            name="default"
            initialParams={initialParams ?? {}}
            component={screen}
          />
        )}
        {children && <Stack.Screen name="default" component={() => children} />}
      </Stack.Navigator>
    </NavigationContainer>
  </ProvideWrapper>
);
