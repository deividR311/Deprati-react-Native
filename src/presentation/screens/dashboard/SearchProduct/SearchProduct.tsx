import React, { useState, useEffect, useMemo } from 'react';
import {
  View,
  Text,
  StyleSheet,
  TextInput,
  Pressable,
  ActivityIndicator,
  KeyboardAvoidingView,
  Platform
} from 'react-native';
import {
  COLORS,
  FONTS_FAMILY,
  FONTS_SIZES,
  NAV
} from '../../../../application/common';
import layout, {
  MARING_HORIZONTAL
} from '../../../../application/common/layout';
import IconMC from 'react-native-vector-icons/MaterialCommunityIcons';
import { useNavigation, useRoute } from '@react-navigation/native';
import { FlatList } from 'react-native-gesture-handler';
import { SafeAreaView } from 'react-native-safe-area-context';
import useDebounce from '../../../../application/common/hooksCommons/useDebounce';
import { removeTags } from '../../../../application/utils/string-formater';
import { useArticleMutationRequest } from '../../../../infrastucture/apis/article';
import { EcommerceNavigationRoute } from '../../../navigation/ecommerce';
import { useLinkPress } from '../../../../application/common/hooksCommons/useLinkPress';
import { useSearchMutationRequest } from '../../../../infrastucture/apis/product';
import NoFoundPage from '../../NoFoundPage/NoFoundPage';

const SearchProduct = () => {
  const navigation = useNavigation();

  const [searchloading, setSearchloading] = useState<boolean>(false);
  const [searchValue, setSearchValue] = useState<string>('');
  const [typeSearch, setTypeSearch] = useState(0);
  const [listProducts, setListProducts] = useState([]);
  const [avoidSearch, setAvoidSearch] = useState(false);
  const [emptySearchArticle, setEmptySearchArticle] = useState(false);
  const [
    _getSearchEan,
    {
      isLoading: loadingSearchEan,
      data: rawResponseSearchEan,
      isError: isErrorSearchEan,
      reset: resetResquestSearchEan
    }
  ] = useArticleMutationRequest();
  const [
    _getSearchProduct,
    {
      isLoading: loadingSearchProduct,
      isUninitialized: uninitializedSearchProduct,
      data: dataSearchProduct,
      reset: resetRequestSearchProduct
    }
  ] = useSearchMutationRequest();

  const route = useRoute();
  const { EAN } = (route?.params as { EAN?: string }) ?? { EAN: '' };
  const { goToPdp } = useLinkPress();

  const onPressEan = () => {
    navigation.navigate(EcommerceNavigationRoute.Barcode as never);
  };

  const getSearch = async (searhText: string) => {
    const request: Promise<any>[] = [];
    const textLen = searhText?.length;
    if (!textLen) return;

    resetRequestSearchProduct();
    resetResquestSearchEan();
    setSearchloading(true);
    setListProducts([]);

    request.push(
      _getSearchProduct({
        query: searhText,
        pageSize: 5,
        fields: 'SEARCH'
      })
    );

    if (textLen >= 8 && !isNaN(Number(searhText))) {
      setEmptySearchArticle(false);
      request.push(
        _getSearchEan({
          codigoTienda: 20,
          eanArticulo: searhText
        })
      );
    }

    await Promise.all(request);
  };

  const noFound = useMemo(() => {
    if (searchValue?.length >= 8 && !isNaN(Number(searchValue))) {
      if (
        listProducts?.length === 0 &&
        (emptySearchArticle || isErrorSearchEan)
      ) {
        //9console.log('00', emptySearchArticle)
        return true;
      }
    } else if (avoidSearch && !searchloading && listProducts?.length === 0) {
      return true;
    }

    return false;
  }, [
    searchValue,
    searchloading,
    listProducts,
    avoidSearch,
    emptySearchArticle
  ]);

  const goToProduct = (item: any) => {
    goToPdp(item?.code);
  };

  const goToSlp = (searhText: string) => {
    if (searhText) {
      navigation.replace(NAV.DASHBOARD_NAVIGATION, {
        screen: NAV.DASHBOARD_INICIO,
        params: {
          screen: NAV.SLP,
          params: { searchID: searhText, title: 'Resultados búsqueda' }
        }
      });
    }
  };

  useEffect(() => {
    if (EAN) {
      try {
        setTypeSearch(1);
        setSearchValue(`${Number(EAN)}`);
      } catch (error) {}
    }
  }, [EAN]);

  const redirectSuccess = () => {
    const products = dataSearchProduct?.products ?? [];
    if (!isNaN(Number(searchValue))) {
      if (products?.length === 0) {
        const { data, success } = rawResponseSearchEan ?? {};

        if (success) {
          const article = data?.dato;
          if (article && article?.listaArticulo?.length > 0) {
            resetResquestSearchEan();
            navigation.navigate(
              EcommerceNavigationRoute.ProductPageSAP as never,
              {
                product: article,
                ean: searchValue
              } as never
            );
          } else {
            setEmptySearchArticle(true);
          }
        }
      }

      if (typeSearch === 1 && products?.length > 0) {
        const [item] = products;
        console.log(item);
        goToProduct(item);
      }

      setListProducts(products);
      setSearchloading(loadingSearchProduct || loadingSearchEan);

      if (!loadingSearchProduct && !loadingSearchEan) {
        setAvoidSearch(!uninitializedSearchProduct);
      }
    } else {
      setListProducts(products);
      setSearchloading(loadingSearchProduct);
      setAvoidSearch(!uninitializedSearchProduct);
    }
  };

  useEffect(() => {
    redirectSuccess();
  }, [
    loadingSearchEan,
    loadingSearchProduct,
    dataSearchProduct?.products,
    uninitializedSearchProduct,
    rawResponseSearchEan?.success
  ]);

  const debouncedSearchTerm = useDebounce(searchValue, 1500);
  useEffect(
    () => {
      setListProducts([]);
      onGetSearchTerm();
    },
    [debouncedSearchTerm] // Only call effect if debounced search term changes
  );

  useEffect(() => {
    if (searchValue?.length > 0) return;
    setEmptySearchArticle(false);
    setAvoidSearch(false);
    setListProducts([]);
    if (!EAN) setSearchValue('');
    resetRequestSearchProduct();
    resetResquestSearchEan();
  }, [searchValue]);

  function onGetSearchTerm() {
    if (debouncedSearchTerm?.length < 3 || debouncedSearchTerm !== searchValue)
      return;
    getSearch(debouncedSearchTerm);
  }

  return (
    <SafeAreaView style={styles.container}>
      <View style={[styles.search]}>
        <Pressable onPress={onPressEan} style={styles.barcode__icon}>
          <IconMC name="barcode-scan" size={24} color={COLORS.DARK70} />
        </Pressable>
        <TextInput
          placeholderTextColor={COLORS.GRAYDARK60}
          style={[styles.search__input, styles.search__input__text]}
          autoFocus={true}
          returnKeyType={'search'}
          onChangeText={text => {
            setSearchValue(text);
          }}
          onSubmitEditing={event => {
            if (listProducts?.length > 0) {
              goToSlp(event.nativeEvent.text);
            } else {
              if (rawResponseSearchEan) {
                redirectSuccess();
              } else {
                onGetSearchTerm();
              }
            }
          }}
          value={searchValue}
        />
        <Pressable onPress={() => navigation.goBack()}>
          <Text style={[styles.search__close]}>Cerrar</Text>
        </Pressable>
      </View>
      <FlatList
        keyExtractor={(item, index) => index.toString()}
        showsHorizontalScrollIndicator={false}
        data={listProducts}
        ListEmptyComponent={() => {
          if (noFound) return <NoFoundPage headerShow={false} />;

          return null;
        }}
        ListHeaderComponent={() => {
          if (searchloading)
            return <ActivityIndicator size="large" color={COLORS.DARK70} />;
          return null;
        }}
        renderItem={({ item, index }: { item: any; index: number }) => {
          return (
            <Pressable key={`item${index}`} onPress={() => goToProduct(item)}>
              <Text style={[styles.search__item]} numberOfLines={1}>
                {removeTags(item.name)}
              </Text>
            </Pressable>
          );
        }}
        style={styles.containerList}
        contentContainerStyle={styles.contentContainerStyle}
      />

      <KeyboardAvoidingView
        behavior={Platform.OS === 'ios' ? 'padding' : undefined}>
        <Pressable
          onPress={() => navigation.replace(EcommerceNavigationRoute.Barcode)}>
          <View style={styles.footerScan}>
            <IconMC
              name="barcode-scan"
              size={24}
              color={COLORS.BRAND}
              style={styles.footerScan_icon}
            />
            <Text style={styles.footerScan_text}>Escanear código</Text>
          </View>
        </Pressable>
      </KeyboardAvoidingView>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    flexDirection: 'column',
    flex: 1,
    backgroundColor: COLORS.WHITE,
    width: '100%',
    justifyContent: 'space-between'
  },
  search__icon: {
    position: 'absolute',
    left: 24,
    bottom: 15,
    zIndex: 2
  },
  barcode__icon: {
    position: 'absolute',
    left: '79%',
    bottom: 13,
    zIndex: 2
  },
  search: {
    display: 'flex',
    position: 'relative',
    backgroundColor: COLORS.WHITE,
    paddingVertical: MARING_HORIZONTAL,
    paddingHorizontal: MARING_HORIZONTAL,
    paddingBottom: 5,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    width: '100%',
    marginTop: layout.isIos ? (layout.isSmallDevice ? -18 : -50) : undefined
  },
  search__input: {
    marginRight: MARING_HORIZONTAL,
    flexGrow: 3,
    paddingLeft: 16,
    paddingRight: 10,
    borderRadius: 3,
    backgroundColor: COLORS.DEPRATYGRAY,
    height: 40,
    maxWidth: '85%'
  },
  search__input__text: {
    fontFamily: FONTS_FAMILY['Roboto-Regular'],
    fontWeight: '400',
    fontSize: FONTS_SIZES.paragraph,
    lineHeight: FONTS_SIZES.paragraph + 2,
    letterSpacing: 0.8,
    color: COLORS.DARK70,
    opacity: 0.7
  },
  search__close: {
    fontSize: FONTS_SIZES.paragraph,
    fontFamily: FONTS_FAMILY['Roboto-Medium'],
    fontWeight: '500',
    lineHeight: FONTS_SIZES.paragraph + 2,
    color: COLORS.DARK70
  },
  contentContainerStyle: {
    backgroundColor: COLORS.WHITE,
    paddingTop: MARING_HORIZONTAL,
    flexGrow: 1
  },
  containerList: {
    width: '100%',
    backgroundColor: COLORS.WHITE
  },
  search__item: {
    paddingHorizontal: MARING_HORIZONTAL,
    paddingVertical: 12,
    fontFamily: FONTS_FAMILY['Roboto-Medium'],
    fontWeight: '500',
    fontSize: FONTS_SIZES.paragraph,
    lineHeight: FONTS_SIZES.paragraph + 2,
    letterSpacing: 0.8,
    color: COLORS.DARK70
  },
  footerScan: {
    backgroundColor: '#FAF9F9',
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'row',
    height: 40,
    borderRadius: 3,
    overflow: 'hidden'
  },
  footerScan_icon: {
    paddingRight: 16
  },
  footerScan_text: {
    color: COLORS.BRAND,
    fontFamily: FONTS_FAMILY['Roboto-Regular'],
    fontWeight: '400',
    fontSize: FONTS_SIZES.paragraph,
    lineHeight: FONTS_SIZES.paragraph + 2
  }
});

export default SearchProduct;
