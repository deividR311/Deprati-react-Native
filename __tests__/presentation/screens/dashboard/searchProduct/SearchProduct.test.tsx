import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationWrapper } from '../../../../../__mocks__/navigation-wrapper';
import { SearchProduct } from '../../../../../src/presentation/screens/dashboard/SearchProduct';
import { mockhookApi } from '../../../../../__mocks__/mock-apis';

jest.mock('../../../../../src/infrastucture/apis/article/article.api', () => ({
  useArticleMutationRequest: () => mockhookApi
}));

let screenTest: any;
const searchProduct_Params = {
  searchID: 'ROPA DEPORTIVA'
};

describe('SearchProduct', () => {
  beforeEach(() => {
    screenTest = render(
      <NavigationWrapper
        initialParams={searchProduct_Params}
        screen={SearchProduct}
      />
    );
  });

  it('renders correctly in Snapshot', () => {
    screenTest = render(<NavigationWrapper screen={SearchProduct} />);
    expect(screenTest.toJSON()).toMatchSnapshot();
  });
});
