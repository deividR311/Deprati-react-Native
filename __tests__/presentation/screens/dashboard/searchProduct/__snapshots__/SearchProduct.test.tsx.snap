// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchProduct renders correctly in Snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        Object {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        hideShadow={true}
        largeTitleHideShadow={false}
        title="default"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            Object {
              "backgroundColor": "#ffffff",
            },
          ]
        }
      >
        <RNCSafeAreaView
          edges={
            Array [
              "bottom",
              "left",
              "right",
              "top",
            ]
          }
          style={
            Object {
              "backgroundColor": "#ffffff",
              "display": "flex",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "space-between",
              "width": "100%",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffffff",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginTop": -50,
                  "paddingBottom": 5,
                  "paddingHorizontal": 16,
                  "paddingVertical": 16,
                  "position": "relative",
                  "width": "100%",
                },
              ]
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "bottom": 13,
                  "left": "79%",
                  "position": "absolute",
                  "zIndex": 2,
                }
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#263238",
                      "fontSize": 24,
                    },
                    undefined,
                    Object {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                󰁲
              </Text>
            </View>
            <TextInput
              autoFocus={true}
              onChangeText={[Function]}
              onSubmitEditing={[Function]}
              placeholderTextColor="#979797"
              returnKeyType="search"
              style={
                Array [
                  Object {
                    "backgroundColor": "#E9EBED",
                    "borderRadius": 3,
                    "flexGrow": 3,
                    "height": 40,
                    "marginRight": 16,
                    "maxWidth": "85%",
                    "paddingLeft": 16,
                    "paddingRight": 10,
                  },
                  Object {
                    "color": "#263238",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 14,
                    "fontWeight": "400",
                    "letterSpacing": 0.8,
                    "lineHeight": 16,
                    "opacity": 0.7,
                  },
                ]
              }
              value=""
            />
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#263238",
                      "fontFamily": "Roboto-Medium",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "lineHeight": 16,
                    },
                  ]
                }
              >
                Cerrar
              </Text>
            </View>
          </View>
          <RCTScrollView
            ListEmptyComponent={[Function]}
            ListHeaderComponent={[Function]}
            collapsable={false}
            contentContainerStyle={
              Object {
                "backgroundColor": "#ffffff",
                "flexGrow": 1,
                "paddingTop": 16,
              }
            }
            data={Array []}
            getItem={[Function]}
            getItemCount={[Function]}
            handlerTag={2}
            handlerType="NativeViewGestureHandler"
            keyExtractor={[Function]}
            onContentSizeChange={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            onLayout={[Function]}
            onMomentumScrollBegin={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            renderScrollComponent={[Function]}
            scrollEventThrottle={50}
            showsHorizontalScrollIndicator={false}
            stickyHeaderIndices={Array []}
            style={
              Object {
                "backgroundColor": "#ffffff",
                "width": "100%",
              }
            }
            viewabilityConfigCallbackPairs={Array []}
          >
            <View>
              <View
                onLayout={[Function]}
              />
            </View>
          </RCTScrollView>
          <View
            onLayout={[Function]}
            style={
              Object {
                "paddingBottom": 0,
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#FAF9F9",
                    "borderRadius": 3,
                    "flexDirection": "row",
                    "height": 40,
                    "justifyContent": "center",
                    "overflow": "hidden",
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#E31A2C",
                        "fontSize": 24,
                      },
                      Object {
                        "paddingRight": 16,
                      },
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  󰁲
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#E31A2C",
                      "fontFamily": "Roboto-Regular",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "lineHeight": 16,
                    }
                  }
                >
                  Escanear código
                </Text>
              </View>
            </View>
          </View>
        </RNCSafeAreaView>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
